src/field/fieldperp.cxx

```diff
@@ -254,3 +341,156 @@ const IndexRange FieldPerp::region(REGION rgn) const {
     break;
   };
 }
+
+//////////////// NON-MEMBER FUNCTIONS //////////////////
```

Comment:
aren't the above also non-member functions?

Reply:
Yes, sorry was thrown by the `FieldPerp::region` which came after non-member stuff.

Reply:
Should be addressed now